### PR TITLE
Fix local strategy and mixed username fields (ie username OR password)

### PR DESCRIPTION
**Pull request by vinzentt**
### Description

When using locale authentication, cookies were never set.

#### Related issues

- #57 and related PR https://github.com/strongloop/loopback-component-passport/pull/88
- #220 

### Checklist

- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by slnode** [comment]
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

File: lib/passport-configurator.js [hunk 1]
```diff
@@ -312,14 +313,7 @@ PassportConfigurator.prototype.configureProvider = function(name, options) {
...
               if (options.setAccessToken) {
-                switch (options.usernameField) {
-                  case  'email':
-                    login({email: username, password: password});
-                    break;
-                  case 'username':
-                    login({username: username, password: password});
-                    break;
-                }
+                login({username: userProfile.username, password: password});
```
Comment (raymondfeng) [inline lib/passport-configurator.js]: Why do you remove the `email` option?

File: lib/passport-configurator.js [hunk 2]
```diff
@@ -491,44 +501,24 @@ PassportConfigurator.prototype.configureProvider = function(name, options) {
               return next(err);
             }
             if (info && info.accessToken) {
+              _setCookies(options, user, info, res, req);
```
Comment (raymondfeng) [inline lib/passport-configurator.js]: This will force the cookie to be set even with `options.json`.

File: lib/passport-configurator.js [hunk 3]
```diff
@@ -538,19 +528,9 @@ PassportConfigurator.prototype.configureProvider = function(name, options) {
   /*
    * Setup the authentication request URLs.
    */
-  if (authType === 'local') {
-    self.app.post(authPath, passport.authenticate(
```
Comment (raymondfeng) [inline lib/passport-configurator.js]: Not sure why passport.authenticate is removed

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Comment by slnode** [comment]
Can one of the admins verify this patch?

**Pull request closed without merging**